lib/offset_gen.rb

```diff
@@ -16,10 +8,10 @@ def last_four_digits(squared_date_param)
   end
 
   def create_offset(last_four_digits_param)#refactor to dynamic
-    { key_A: last_four_digits_param[0].to_i,
-      key_B: last_four_digits_param[1].to_i,
-      key_C: last_four_digits_param[2].to_i,
-      key_D: last_four_digits_param[3].to_i
-    }
+    [ last_four_digits_param[0].to_i,
+      last_four_digits_param[1].to_i,
+      last_four_digits_param[2].to_i,
+      last_four_digits_param[3].to_i
+    ]
```

Comment:
was originally using a hash here, but an array was the better choice to implement encryption